src/volttron/utils/dynamic_helper.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import annotations
+
+import importlib
+import inspect
+import logging
+from types import ModuleType
+from typing import List, Type
+
+__all__: List[str] = ["get_module", "get_klass", "get_subclasses"]
+
+_log = logging.getLogger(__name__)
+
+
+def get_module(module: str) -> ModuleType:
+    """Returns a dynamically loaded module. If not found on pythonpath, then raise a ModuleNotFound error.
+    This method is a wrapper around Python's builtin function, importlib.import_module(...).
+    See https://docs.python.org/3/library/importlib.html#importlib.import_module
+
+    :param module: The name argument specifies what module to import in absolute or relative terms (e.g. either pkg.mod or ..mod).
+        If the name is specified in relative terms, then the package argument must be set to the name of the package which is to act as the
+        anchor for resolving the package name (e.g. import_module('..mod', 'pkg.subpkg') will import pkg.mod).
+    :type module: str
+    :raises ModuleNotFoundError:
+    :return: A module
+    :rtype: ModuleType
+    """
+    try:
+        return importlib.import_module(module)
+    except ModuleNotFoundError as e:
+        _log.debug(f"Module: {module} not found. Make sure it is on the PYTHONPATH")
+        raise e
+
+
+def get_klass(module: str | ModuleType, klass_name: str) -> Type:
+    """Retrieve a Type from a module. If module is a string, attempt to load it via importlib.import_module.
+    If not a string, then directly look for a class within the passed module.
+
+    :param module: the path to a module or the actual module
+    :type module: str | ModuleType
+    :param klass_name: the name of the type in the module
+    :type klass_name: str
+    :raises AttributeError:
+    :return: Returns the class from the module
+    :rtype: Type
+    """
+    try:
+        if isinstance(module, str):
+            return getattr(get_module(module), klass_name)
+        return getattr(module, klass_name)
+    except AttributeError as e:
+        _log.debug(f"Class {klass_name} is not defined in {module}.")
+        raise e
+
+
+def get_subclasses(module: ModuleType, parent_klass: Type, return_all=False) -> List[Type]:
```

Comment:
Let's allow a string and type for the module and parent_class.

So

ModuleType | str 
and Type | str

Then we can use the other functions that do checking etc.

Reply:
Ok.